common/json/platform/nlohmann/nlohmann_json.cpp

```diff
@@ -79,7 +79,13 @@ ExpectedJson Load(istream &str) {
 
 ExpectedJson Load(io::Reader &reader) {
 	auto str_ptr = reader.GetStream();
-	return Load(*(str_ptr.get()));
+	try {
+		return Load(*str_ptr);
+	} catch (njson::parse_error &e) {
```

Comment:
I suppose we could catch `njson::exception` instead. It seems to be the top-level exception it uses, and it inherits from `std::exception`. Any higher exceptions that we should not catch I think, because it's not supposed to throw them, so if it happens, something is very wrong with the code, and it's better that we know about it.

This is pre-existing, so I can apply in a new commit.

Reply:
@oleorhagen: Well, I need to eat my words here. Turns out JSON *does* throw other exceptions. Or rather, it doesn't catch exceptions from `std`. So on an open error, these escape out of the JSON library.

I'll change this in an upcoming commit.

Reply:
Good god - my spidey senses end up victorious this time!